shared/src/state-migrations/19-rename-start-point-types.ts

```diff
@@ -0,0 +1,143 @@
+import type { UUID } from '../utils';
+import type { Migration } from './migration-functions';
+
+export const renameStartPointTypes19: Migration = {
+    actions: (_initialState, actions) => {
+        actions.forEach((action) => {
+            if (
+                (action as { type: string } | null)?.type ===
+                '[Transfer] Add to transfer'
+            ) {
+                const typedAction = action as {
+                    startPoint: {
+                        type:
+                            | 'alarmGroup'
+                            | 'alarmGroupStartPoint'
+                            | 'transferPoint'
+                            | 'transferStartPoint';
+                    };
+                };
+                if (typedAction.startPoint.type === 'alarmGroup') {
+                    typedAction.startPoint.type = 'alarmGroupStartPoint';
+                } else if (typedAction.startPoint.type === 'transferPoint') {
+                    typedAction.startPoint.type = 'transferStartPoint';
+                }
+            }
+        });
+    },
+    state: (state) => {
+        const typedState = state as {
+            materials: {
+                [materialId: UUID]: {
+                    position:
+                        | {
+                              type: 'transfer';
+                              transfer: {
+                                  startPoint: {
+                                      type:
+                                          | 'alarmGroup'
+                                          | 'alarmGroupStartPoint'
+                                          | 'transferPoint'
+                                          | 'transferStartPoint';
+                                  };
+                              };
+                          }
+                        | { type: 'coordinates' }
+                        | { type: 'simulatedRegion' }
+                        | { type: 'vehicle' };
```

Comment:
I think you can omit these other possible position types on materials, vehicles and personnel as they're not relevant for the migration.

Reply:
I didn't want to put position: any as a possibility and if I put {type: Exclude<any,'transfer'}> as a possibility for position to put the 3 cases in one it doesn't work either, so this is the only way I could think of w/o putting any as an option for the transfer Position as well.

Reply:
If you know a better way, I am open though

Reply:
You don't have to specify the other possible types at all since you don't care about them. Just use

```ts
[materialId: UUID]: {
    position: {
        type: 'transfer',
        transfer: {
            startPoint: {
                type:
                    | 'alarmGroup'
                    | 'alarmGroupStartPoint'
                    | 'transferPoint'
                    |  'transferStartPoint';
            };
        };
    }
}
```

Or am I missing something here?

Reply:
Hmm, It works which I think is weird because now I am saying that the type is always transfer AND I'm checking it in the if statement later. In my mind, this should lead to a Condition is always true warning, but it doesn't.